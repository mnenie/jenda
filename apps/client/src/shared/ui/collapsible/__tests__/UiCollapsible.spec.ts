import { beforeEach, describe, expect, it } from 'vitest'
import { mount, type VueWrapper } from '@vue/test-utils'
import TestCollapsible from './MockComponent.vue'

describe('tests for UiCollapsible', () => {
  let wrapper: VueWrapper<InstanceType<typeof TestCollapsible>>
  beforeEach(() => {
    wrapper = mount(TestCollapsible, {
      props: {
        defaultOpen: true,
      },
    })
  })

  it('should render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit events correctly', () => {
    wrapper.vm.$emit('update:open', true)
    expect(wrapper.emitted('update:open')).toBeTruthy()
    expect(wrapper.emitted('update:open')?.[0]).toEqual([true])
  })
})
