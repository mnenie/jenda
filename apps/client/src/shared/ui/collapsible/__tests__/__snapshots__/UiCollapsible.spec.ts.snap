// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for UiCollapsible > should render correctly 1`] = `
"<div data-state="closed"><button type="button" aria-controls="" aria-expanded="false" data-state="closed"> trigger </button>
  <div class="collapsible-content" id="radix-vue-collapsible-content-v-0" data-state="closed" hidden="" style="--radix-collapsible-content-height: 0px; --radix-collapsible-content-width: 0px;">
    <!---->
  </div>
</div>"
`;
